### PR TITLE
Move Runner to its own class to ease testing

### PR DESCRIPTION
**Pull request by dkjellin**
Moved the Runner out to its own class and updated tests to not need sleeps etc to test the functionality.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -223,10 +230,6 @@
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-surefire-plugin</artifactId>
-                        <configuration>
-                            <!-- Skip the unit tests when the it profile is turned on -->
-                            <skip>true</skip>
-                        </configuration>
```
Comment (justinskariah) [inline pom.xml]: Do we want to run unit tests along with integration tests? If yes then we need to update the bamboo builds to reflect the same

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketTriggerWorker.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            PollingResult result = triggerItem.poll(listener);
+            logger.println("Poll complete. Took " + Util.getTimeSpanString(System.currentTimeMillis() - start));
+
+            if (result.hasChanges()) {
+                logger.println("Changes since last build: " + result.change);
+                triggerItem.scheduleBuild2(0, actions.toArray(ACTION_ARRAY));
+            }
+        } catch (IOException | RuntimeException e) {
+            LOGGER.log(Level.SEVERE, String.format(
+                    "Failed to trigger job %s because an error occurred while writing the polling log to %s",
```
Comment (justinskariah) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketTriggerWorker.java]: Not part of this change, but can `RuntimeException` and `IOException` only be caused by log errors? Can `triggerItem.poll()` throw exception?

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerCause.java [hunk 3]
```diff
@@ -12,11 +14,29 @@ public BitbucketWebhookTriggerCause(BitbucketWebhookTriggerRequest triggerReques
...
     public String getShortDescription() {
         return triggerRequest
                 .getActor()
                 .map(actor -> BitbucketWebhookTriggerCause_withAuthor(actor.getDisplayName()))
                 .orElseGet(Messages::BitbucketWebhookTriggerCause_withoutAuthor);
     }
+
+    @Override
+    public int hashCode() {
+
```
Comment (justinskariah) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerCause.java]: empty line

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java [hunk 4]
```diff
@@ -84,83 +77,8 @@ public void schedule(
...
-                }
-            } catch (IOException | RuntimeException e) {
-                LOGGER.log(Level.SEVERE, String.format(
-                        "Failed to trigger job %s because an error occurred while writing the polling log to %s",
-                        job,
-                        logFile.getPath()),
-                        e);
-            }
+            CauseAction causeAction = new CauseAction(new BitbucketWebhookTriggerCause(triggerRequest));
+            queue.execute(new BitbucketTriggerWorker(job, triggerItem, causeAction, triggerRequest.getAdditionalActions()));
```
Comment (justinskariah) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookTriggerImpl.java]: nit: pass `triggerRequest` into `BitbucketTriggerWorker` and create `CauseAction` in `BitbucketTriggerWorker` constructor

File: pom.xml (hunk 1 above)
Comment (dkjellin) [reply pom.xml]: Yes, that is the norm. It makes sense to me. I can run `mvn verify` to run _just_ unit tests, I can also run `mvn -Pit verify` to run _all_ the tests. If we do not do this you'd have to do two invocations to run all tests, which seems suboptimal.

I don't think we necessarily need to update builds, if we feel that unit tests unnecessarily slows down the integration test run in the build system we can add properties to disable just unit tests.
